UI/src/app/services/upload.service.ts

```diff
@@ -36,6 +36,7 @@ export class UploadService {
       );
       this.bridgeService.resetAllMappings();
       this.dataService.prepareTables(res, 'source');
+      this.dataService.saveReportName(files[0].name.substr(0, files[0].name.lastIndexOf('.')), 'report')
```

Comment:
move position of dot into separate variable

Let's use slice. Slice is more universal
substr is supported by historical reasons(also substr is not described in JS specification itself)